pallets/pallet-governance/src/lib.rs

```diff
@@ -0,0 +1,407 @@
+				Votes::<T>::remove(proposal);
+			});
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Registered a proposal.
+		///
+		/// Require a member from council.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::initiate_proposal())]
+		pub fn initiate_proposal(
+			origin: OriginFor<T>,
+			call: <T as Config>::RuntimeCall,
+		) -> DispatchResult {
+			let id = ensure_signed(origin)?;
+			ensure!(CurrentAuthorities::<T>::get().contains(&id), Error::<T>::InvalidAuthority);
+			let proposal_id = Self::next_proposal_id();
+			let expired_block =
+				T::BlockNumberProvider::current_block_number() + T::ExpiryPeriod::get();
+
+			// add call to storage
+			Proposals::<T>::insert(proposal_id, call.encode());
+
+			// add proposal id to expiry.
+			// Get the BTreeSet for the block number or create a new one if it doesn't exist
+			let mut expiry_set = Expiry::<T>::get(expired_block).unwrap_or_default();
+
+			// Insert the proposal id into the set
+			expiry_set.insert(proposal_id);
+
+			// Store the updated set back into storage
+			Expiry::<T>::insert(expired_block, expiry_set);
+
+			// add vote to Votes
+			let mut set = BTreeSet::new();
+			set.insert(id.clone());
+			Votes::<T>::insert(proposal_id, CastedVotes { yays: set, nays: Default::default() });
+
+			// Emit Event: ProposalRegistered
+			Self::deposit_event(Event::<T>::ProposalRegistered { who: id, call });
+
+			Ok(())
+		}
+
+		// Requires Root origin - sets the current authorities
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::vote())]
+		pub fn vote(origin: OriginFor<T>, proposal: ProposalId, approve: bool) -> DispatchResult {
+			// check if the vote can be casted
+			// is member of the council
+			let id = ensure_signed(origin)?;
+			ensure!(CurrentAuthorities::<T>::get().contains(&id), Error::<T>::InvalidAuthority);
+
+			// Proposal is valid
+			ensure!(Proposals::<T>::get(proposal).is_some(), Error::<T>::InvalidProposalId);
+
+			if let Some(mut casted) = Votes::<T>::get(proposal) {
+				// has not voted
+				ensure!(
+					!casted.yays.contains(&id) && !casted.nays.contains(&id),
+					Error::<T>::AlreadyVoted
+				);
+
+				// Cast the vote
+				if approve {
+					casted.yays.insert(id.clone());
+				} else {
+					casted.nays.insert(id.clone());
+				}
+
+				// check if the proposal is ready to be resolved.
+				let count_yays = casted.yays.len();
+				let count_nays = casted.nays.len();
+				let count_council = CurrentAuthorities::<T>::get().len();
+				let pass: Percent = Percent::from_rational(count_yays as u32, count_council as u32);
+				let fail: Percent = Percent::from_rational(count_nays as u32, count_council as u32);
+
+				if pass >= T::MajorityThreshold::get() {
+					ProposalsToResolve::<T>::get().insert((proposal, true));
+				}
+
+				if fail >= Percent::one().saturating_sub(T::MajorityThreshold::get()) {
+					ProposalsToResolve::<T>::get().insert((proposal, false));
+				}
+			} else {
```

Comment:
If this is impossible we can just ignore this codepath.

Reply:
It is possible to be here after rotating authority. We need to treat the vote as valid here